R/strat_ests_barplot.R

```diff
@@ -0,0 +1,86 @@
+#' Barplot method for `summary.seroincidence.by` objects
+#'
+#' @param object a `summary.seroincidence.by` object (generated by applying the
+#' `summary()` method to the output of [estimate_scr_by()]).
+#' @param yvar the name of a stratifying variable in `object`.
+#' @param color_var
+#' [character] the name of the fill color variable (e.g., "Country").
+#' @param alpha
+#' transparency for the bars (1 = no transparency, 0 = fully transparent).
+#' @param CIs [logical], if `TRUE`, add CI error bars.
+#' @param title a title for the final plot.
+#' @param xlab a label for the x-axis of the final plot.
+#' @param ylab a label for the y-axis of the final plot.
+#' @param fill_lab fill label.
+#' @param fill_palette optional color palette for bar fill.
+#' @param ... unused.
+#'
+#' @return a [ggplot2::ggplot()] object.
+#' @export
+#' @keywords internal
+#'
+strat_ests_barplot <- function(
+    object,
+    yvar,
+    color_var = NULL,
+    alpha = 0.7,
+    CIs = FALSE, # nolint: object_name_linter
+    title = NULL,
+    xlab = "Seroconversion rate per 1000 person-years",
+    ylab = yvar,
+    fill_lab = NULL,
+    fill_palette = NULL,
+    ...) {
+
+  # Check if yvar exists in the dataset
+  if (!is.element(yvar, names(object))) {
+    cli::cli_abort(
+      class = "unavailable_yvar",
+      message = c(
+        "The variable `{yvar}` specified by argument `yvar`
+        does not exist in `object`.",
+        "Please choose a column that exists in `object`."
+      )
+    )
+  }
+
+  plot1 <- ggplot2::ggplot(object, ggplot2::aes(
+    y = forcats::fct_rev(.data[[yvar]]),
+    x = .data$incidence.rate * 1000,
+    fill = if (!is.null(color_var)) .data[[color_var]] else NULL
+  )) +
```

Comment:
When possible, please separate the `ggplot()` call and the `aes()` call, like this:

```suggestion
  plot1 <- ggplot2::ggplot(object) + 
  ggplot2::aes(
    y = forcats::fct_rev(.data[[yvar]]),
    x = .data$incidence.rate * 1000,
    fill = if (!is.null(color_var)) .data[[color_var]] else NULL
  ) +
```

Reply:
@d-morrison When I separated `ggplot()` and `aes()` as you suggested I kept running into layers of errors from unrecognized datasets and uninherited aesthetics. I'd like to revert it to the original code, unless you have other suggestions?